pages/api/registrobr/v1/[domain].js

```diff
@@ -0,0 +1,21 @@
+import app from '@/app';
+import BadRequestError from '@/errors/BadRequestError';
+import NotFoundError from '@/errors/NotFoundError';
+import getRegistroBrAvail from '@/services/registro';
+
+async function data(request, response) {
+  try {
+    const result = await getRegistroBrAvail(request.query.domain);
```

Comment:
O que acha de validar o domínio antes de fazer a chamada para a API?

[sugestão de implementação](https://www.geeksforgeeks.org/how-to-validate-a-domain-name-using-regular-expression/)